test/tests/problems/eigen_problem/jfnk_mo/tests

```diff
@@ -0,0 +1,44 @@
+[Tests]
+  design = "Eigenvalue.md"
+  issues = "#18493"
+
+  [ne_array_mo]
+    type = 'CSVDiff'
+    input = 'ne_array_mo.i'
+    csvdiff = 'ne_array_mo_out_eigenvalues_0001.csv'
+    slepc = true
+    slepc_version = '>=3.11.0'
+    expect_out = 'PJFNK\s+with\s+matrix\s+only'
+    requirement = "The system shall support use of matrix-vector multiplication as residual evaluation for eigenvalue calculations"
+  []
+
+  [ne_coupled_mo]
+    type = 'CSVDiff'
+    input = 'ne_coupled_mo.i'
+    csvdiff = 'ne_coupled_eigenvalues_0001.csv'
+    slepc = true
+    slepc_version = '>=3.11.0'
+    requirement = "The system shall support use of matrix-vector multiplication as residual evaluation for coupled eigenvalue problems"
```

Comment:
What would happen if a user did not have `full = true` in their preconditioning block? Would we error? Because then the residual evaluation that this is trying to replace would not be accurate, correct?

Reply:
It depends. Users might have a slow convergence (or even divergence), but answer will be wrong because we have Jacobian/residual on every nonlinear iteration